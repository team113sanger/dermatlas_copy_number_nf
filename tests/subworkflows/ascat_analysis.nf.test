nextflow_workflow {

    name "Test Workflow ASCAT_ANALYSIS"
    script "subworkflows/ascat_analysis.nf"
    workflow "ASCAT_ANALYSIS"

    test("Should run without failures") {

        when {
            params {
                load("$baseDir/tests/testdata/test_params.json")
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = tuple(
                [pair_id: "PD54403a_PD54403b", normal: "PD54403a", tumor: "PD54403b", Sex: "M", Phenotype: "N", sexchr: "XY"], 
                file("/lustre/scratch124/nf-test/copy_number/bams/PD54403a/PD54403a.sample.dupmarked.bam"), 
                file("/lustre/scratch124/nf-test/copy_number/bams/PD54403a/PD54403a.sample.dupmarked.bam.bai"), 
                file("/lustre/scratch124/nf-test/copy_number/bams/PD54403b/PD54403b.sample.dupmarked.bam"),
                file("/lustre/scratch124/nf-test/copy_number/bams/PD54403b/PD54403b.sample.dupmarked.bam.bai"))
                input[1] = "${params.OUTDIR}"
                input[2] = file("${params.reference_genome}")
                input[3] = file("${params.bait_set}") 
                input[4] = file("${params.resource_files}")
                input[5] = file("${params.gc_file}")
                input[6] = file("${params.rt_file}")
                input[7] = "test"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
