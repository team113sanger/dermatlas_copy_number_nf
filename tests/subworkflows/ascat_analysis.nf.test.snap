{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "one_patient_per_tumor.txt:md5,21397cd1b7cf37532c785939e6b18cc6"
                ],
                "1": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "ASCAT_estimates_PD54403b_PD54403a.tsv:md5,a6a11dcbc68bf08b6d2129aa4f984786"
                    ]
                ],
                "segment_summary": [
                    "one_patient_per_tumor.txt:md5,21397cd1b7cf37532c785939e6b18cc6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-28T14:54:53.497924"
    }
}